### DISCUSSION TITLE
Link input field and "button" so that "button"'s onClick calls a function with the input field's event

### DISCUSSION THREAD

**Discussion by nora-kauczor** [neuefische/questions#832] [Web] [answered]
Hello,

we are currently working on a form inside our app. 
The user inserts a recipe for a remedy, there.
Before submitting his choices, the user will see the single ingredients for his remedy and the symptoms that it treats displayed below the input fields (see right side of picture below):

![Image](https://github.com/Larissartoricader/Capstone/assets/160526052/eee6416c-0e84-407f-83e2-bd9333669a98)

- In order to have a better usability for mobile use, we want the user to be able to "append" ingredients and symptoms (= have the displayed below the input fields without actually submitting) via a button (instead of e.g. by pressing a certain key)
- The "button" doesn't have to necessarily be an actual `button` element, but could be any html element made clickable by adding `style={{cursor: "pointer",}}` (This might even be better since a `button` element would require the user to fill all required fields of the form before it does anything.)
- The button/clickable html element is right after e.g. the ingredient input field inside the JSX.
- This is because, unfortunately, it can't be implement inside the input field. <- **Is this true?**
- The button/clickable html element receives an onClick property. When it is clicked, it calls a function `handleClick` that receives `event` as an input. 
- We need `event` to be the user input from an input field, e.g. from the input field for ingredients.
**Is there a way for us to link the clickable html element and the input field?** Or is there any other solution.

You can have a look at the form [here](https://github.com/Larissartoricader/Capstone/blob/main/components/RecipeForm.js), but there is no such button implement by now. 

Thank you for your help!

I'm aware this post doesn't have the demanded structure, but due to group work and several branches and waiting time it was not possible to code a "try" of what is described above in order to show it to the reader. Still, we have of course tried it before, otherwise we wouldn't have had the problem.

**Comment by MareikeFla** [comment]
Hi :) 
if I understand correctly what you have in mind you can achieve this by creating a state for the input field value and then passing that value to the function called by the button's onClick event


```
const [inputValue, setInputValue] = useState('');

  function handleInputChange (event) {
    setInputValue(event.target.value);
  };

  function handleButtonClick () {
    // Call your function with the inputValue
    // Reset input field if needed
    setInputValue('');
  };

  return (

    <div>
      <input type="text" value={inputValue} onChange={handleInputChange} />
      <button onClick={handleButtonClick}>Submit</button>
    </div>

  );
```

**Comment by alexxandar** [comment, marked as answer]
> * This is because, unfortunately, it can't be implement inside the input field. <- **Is this true?**

Yes, input field cannot have any HTML elements inside of it. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
> * We need `event` to be the user input from an input field, e.g. from the input field for ingredients.

If you want the event to be user input from a field, it sounds like the change event is what you need. 

- https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event
- https://upmostly.com/tutorials/react-onchange-events-with-examples

**Reply by nora-kauczor** [reply]
Thank you :) I solved it with onChange:

```
import React from "react";
import { useState } from "react";

export default function TestFunction() {
  const [appended, setAppended] = useState();
  const [userInput, setUserInput] = useState();
  function handleClick() {
    setAppended(userInput);
  }

  return (
    <>
      <form>
        <input
          onChange={() => setUserInput(event.target.value)}
        ></input>
        <p
          style={{
            cursor: "pointer",
          }}
          onClick={handleClick}
        >
          Click me
        </p>
        <div>{appended}</div>
      </form>
    </>
  );
}
```
